### PR TITLE
update(config): clarify deprecation notices + list all env vars

### PR DESCRIPTION
**Pull request by incertum**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

> /kind release

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

/area engine

> /area tests

> /area proposals

> /area CI

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

@FedeDP included https://github.com/falcosecurity/falco/issues/2949

**Does this PR introduce a user-facing change?**:

```release-note
update(config): clarify deprecation notices + list all env vars
```

### CODE REVIEW

File: falco.yaml [hunk 1]
```diff
@@ -977,7 +992,7 @@ metrics:
 # Falco performance tuning (advanced) #
 #######################################
 
-# [DEPRECATED] `syscall_buf_size_preset`
+# [DEPRECATED] `syscall_buf_size_preset` -> Replaced by `engine.<driver>.buf_size_preset` starting Falco 0.38!
```
Comment (incertum) [inline falco.yaml:995]: @LucaGuerra and @FedeDP minor follow up to inform adopters better.

**Review by Andreagit97 — APPROVED** [review]
/approve

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**